### PR TITLE
libswoc: Update Errata use in cache tool.

### PR DESCRIPTION
**Pull request by SolidWallOfCode**

### CODE REVIEW

**Comment by SolidWallOfCode** [comment]
```
/home/jenkins/workspace/Github_Builds/rocky/src@tmp/durable-351524f7/script.sh: line 4: /opt/rh/gcc-toolset-11/enable: No such file or directory
```

**Comment by bneradt** [comment]
[approve ci rocky]

File: src/traffic_cache_tool/CacheTool.cc [hunk 1]
```diff
@@ -1424,7 +1424,7 @@ main(int argc, const char *argv[])
     arguments.invoke();
   }
 
-  if (err.size()) {
+  if (err.length()) {
```
Comment (bneradt) [inline src/traffic_cache_tool/CacheTool.cc:1427]: Why a length check rather than if (!err.is_ok()) {

**Review by bneradt — CHANGES_REQUESTED** [review]

**Review by bneradt — DISMISSED** [review]

**Comment by bneradt** [comment]
[approve ci clang-format]

File: include/tscpp/util/ts_errata.h [hunk 2]
```diff
@@ -40,5 +40,16 @@ static constexpr std::array<swoc::TextView, 9> Severity_Names{
   {"Diag", "Debug", "Status", "Note", "Warn", "Error", "Fatal", "Alert", "Emergency"}
 };
 
+inline std::error_code
+ec_for()
+{
+  return {errno, std::system_category()};
```
Comment (jpeach) [inline include/tscpp/util/ts_errata.h:46]: Is this right? The docs say that errno is in generic_category

https://en.cppreference.com/w/cpp/error/generic_category

File: include/tscpp/util/ts_errata.h [hunk 3]
```diff
@@ -40,5 +40,16 @@ static constexpr std::array<swoc::TextView, 9> Severity_Names{
...
+}
+inline std::error_code
+ec_for(int err)
+{
+  return {err, std::system_category()};
+}
+
```
Comment (jpeach) [inline include/tscpp/util/ts_errata.h:53]: Nit: the name `ec_for` isn't very intuitive. IMHO naming this consistent with `make_error_code` would be more easily understandable.

File: include/tscpp/util/ts_errata.h (hunk 2 above)
Comment (SolidWallOfCode) [reply include/tscpp/util/ts_errata.h:46]: I considered that, but in practice `errno` is system specific (e.g., the value can vary between operating systems) and therefore, IMHO, should be `system_category`. If we look at the [documentation for system_category](https://en.cppreference.com/w/cpp/error/system_category) we see

"Obtains a reference to the static error category object for errors reported by the operating system", which is precisely what `errno` is.

File: include/tscpp/util/ts_errata.h (hunk 3 above)
Comment (SolidWallOfCode) [reply include/tscpp/util/ts_errata.h:53]: Changed to `make_errno_code`.

File: include/tscpp/util/ts_errata.h (hunk 2 above)
Comment (jpeach) [reply include/tscpp/util/ts_errata.h:46]: And that that text continues:
```
It is also required to override the virtual function [std::error_category::default_error_condition()](https://en.cppreference.com/w/cpp/error/error_category/default_error_condition) to map the error codes that match POSIX [errno](https://en.cppreference.com/w/cpp/error/errno) values to [std::generic_category](https://en.cppreference.com/w/cpp/error/generic_category).
```

So, this means that system category is required to forward errno values to generic category? TBH this error abstraction seems to provide more confusion that value 🤷

**Review by jpeach — DISMISSED** [review]

**Comment by SolidWallOfCode** [comment]
Had to update due to conflicts.

**Review by bneradt — APPROVED** [review]

**Pull request merged**